Update to CREST 3.0.2

pprcht opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* ~~[ ] Bumped the build number (if the version is unchanged)~~
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->
Tested locally with conda-build 24.7.1, license file is being packed.
`tblite` is currently *not* in the `requirements` because the git submodule points to a fork. Will be changed in future versions of the code.

pprcht commented:
@conda-forge-admin, please rerender

awvwgk reviewed (changes requested):
We cannot use git sources here and certainly not git submodules as those would make it challenging to have reproducible package builds.

In general, if you have dependencies from git submodules please define multiple sources and declare all dependencies explicitly.

Finally, for Lapack/Blas the liblapack/libblas dependencies should be used to allow cf to seamlessly switch between openblas and mkl. One issue I encountered with liblapack/libblas is that the come with pkgconfig files which break the logic for detecting those libs in some build systems. A solution is to clobber the pkgconfig files in the build step and restore them after the package is built.

pprcht commented:
Okay, fair enough, I can do the individual sources instead of the git submodules.

Any advice on how to do the libblas/lapack pkgconfig thing? Just moving them manually in the build step before/after the cmake stuff?

pprcht commented:
I added all dependencies explicitly to make everything reproducible and reverted to the use of only libblas/liblapack to avoid the pkgconfig issue.

There might be an annoying warning from OpenBLAS  when running the program when it was build this way, which can be circumvented by setting a global variable (`OPENBLAS_NUM_THREADS`). I read in the conda-forge documentation that it is possible to add `activate.sh`/`deactivate.sh` scripts for such purpose, which might be a way to go here and set the global variable? Otherwise, if that's an ugly solution, I will just point this out in the program documentation.

I don't quite get why the Azure osx_arm64_ build failed there, this is just one of the standard CMake checks (which also did *not* fail in my last iteration of the PR, and also did *not* fail on the osx_64_ build). Is there a way to re-run it?

pprcht commented:
Thanks for looking into the test-drive cross-compilation & dependencies!
